iidm/iidm-modification/src/test/java/com/powsybl/iidm/modification/ConnectGeneratorTest.java

```diff
@@ -101,4 +107,67 @@ void testConnectGeneratorWithNoNetworkInformation() {
+        checkInitalStatusTerminalDisconnected(network, generatorsToConnectTerminal);
+
+        // These generators are linked to grid through a line that is disconnected: it should remain that way
+        List<String> generatorsRemainsDisconnectedToTheMainComponent = List.of("ESCDGU1 _generator", "ESCDGN1 _generator");
+
+        // These generators are connected to grid through 2 transformers that are disconnected: they should be reconnected
+        List<String> generatorsToConnectToTheMainComponent = List.of("ESD2GU1 _generator", "ESDTGU1 _generator", "ESD2GN1 _generator", "ESDTGN1 _generator");
+        checkInitialStatusDisconnected(network, Stream.concat(generatorsToConnectToTheMainComponent.stream(), generatorsRemainsDisconnectedToTheMainComponent.stream()).collect(Collectors.toList()));
+
+        for (List<String> strings : Arrays.asList(generatorsInitiallyConnected, generatorsToConnectTerminal, generatorsToConnectToTheMainComponent, generatorsRemainsDisconnectedToTheMainComponent)) {
+            for (String id : strings) {
+                new ConnectGenerator(id).apply(network);
+            }
+        }
+        checkTerminalConnectedButNotInMain(network, generatorsRemainsDisconnectedToTheMainComponent);
+        checkInitialStatusConnected(network, generatorsInitiallyConnected);
+        checkInitialStatusConnected(network, Stream.concat(generatorsToConnectToTheMainComponent.stream(), generatorsToConnectTerminal.stream()).collect(Collectors.toList()));
+
+    }
+
+    private void checkTerminalConnectedButNotInMain(Network network, List<String> ids) {
+        ids.forEach(id -> {
+            Generator gen = network.getGenerator(id);
+            assertTrue(gen.getTerminal().isConnected());
+            assertFalse(gen.getTerminal().getBusBreakerView().getConnectableBus().isInMainSynchronousComponent());
+        });
+    }
+
+    private void checkInitialStatusConnected(Network network, List<String> ids) {
+        ids.forEach(id -> {
+            Generator gen = network.getGenerator(id);
+            assertTrue(gen.getTerminal().isConnected());
+            assertTrue(gen.getTerminal().getBusBreakerView().getConnectableBus().isInMainSynchronousComponent());
+        });
+    }
+
+    private void checkInitialStatusDisconnected(Network network, List<String> ids) {
+        ids.forEach(id -> {
+            Generator gen = network.getGenerator(id);
+            assertFalse(gen.getTerminal().isConnected());
+            assertFalse(gen.getTerminal().getBusBreakerView().getConnectableBus().isInMainSynchronousComponent());
+        });
+    }
+
+    private void checkInitalStatusTerminalDisconnected(Network network, List<String> ids) {
+        ids.forEach(id -> {
+            Generator gen = network.getGenerator(id);
+            assertFalse(gen.getTerminal().isConnected());
+            assertTrue(gen.getTerminal().getBusBreakerView().getConnectableBus().isInMainSynchronousComponent());
+        });
+    }
```

Comment:
You might want to replace those 4 methods with one:
```suggestion
    private void checkTerminalStatus(Network network, List<String> ids, boolean shouldBeConnected, boolean shouldBeInMainSynchronousComponent) {
        ids.forEach(id -> {
            Generator gen = network.getGenerator(id);
            assertEquals(shouldBeConnected, gen.getTerminal().isConnected());
            assertEquals(shouldBeInMainSynchronousComponent, gen.getTerminal().getBusBreakerView().getConnectableBus().isInMainSynchronousComponent());
        });
    }
```
and fix the call to this method in the test